molecularnodes/entities/molecule/reader.py

```diff
@@ -0,0 +1,196 @@
+from abc import ABCMeta
+from pathlib import Path
+from io import BytesIO
+from biotite.structure import AtomArray, AtomArrayStack
+from biotite import structure as struc
+import numpy as np
+from ...assets import data
+from ... import color
+
+
+class ReaderBase(metaclass=ABCMeta):
+    """
+    Abstract base class for reading molecular data from a file.
+    """
+
+    def __init__(self, file_path: str | Path | BytesIO):
+        self.file_path = file_path
+        self.file = self.read(file_path)
+        self.array = self.get_structure()
+        self.set_standard_annotations()
+
+    def read(self, file_path):
+        raise NotImplementedError("Subclasses must implement this method.")
+
+    def set_standard_annotations(self):
+        annotations = {
+            "mass": _compute_mass,
+            "atomic_number": _compute_atomic_number,
+            "res_name_int": _compute_res_name_int,
+            "chain_id_int": _compute_chain_id_int,
+            "vdw_radii": _compute_vdw_radii,
+            "atom_name_int": _compute_atom_name_int,
+            "charge": _compute_charge,
+            "lipophobicity": _compute_lipophobicity,
+            "Color": _compute_color,
+            "is_alpha_carbon": _compute_is_alpha_carbon,
+            "is_solvent": _compute_is_solvent,
+            "is_backbone": _compute_is_backbone,
+            "is_nucleic": _compute_is_nucleic,
+            "is_peptide": _compute_is_peptide,
+            "is_side_chain": _compute_is_side_chain,
+            "is_carb": _compute_is_carb,
+        }
+        for key, func in annotations.items():
+            try:
+                self.array.set_annotation(key, func(self.array))
+            except Exception:
+                pass
+
+    def get_structure(self) -> AtomArrayStack | AtomArray:
+        raise NotImplementedError("Subclasses must implement this method.")
+
+
+def _compute_mass(array):
```

Comment:
style suggestion: add all of these methods to the class.  Are they used elsewhere?